### ISSUE TITLE
Monolingual 1.7.6 appears to delete files from /private/var/root/.Trash

### ISSUE THREAD

**Issue by SeanMSmith** [IngmarStein/Monolingual#106]
I just ran Monolingual 1.7.6 for the first time and noticed the following new behaviors (which I'm reporting together in case they're all manifestations of the same issue, as seems probable).

I'm running macOS Sierra 10.12.3.  (I last ran Monolingual 1.7.3 in El Capitan.)

Note that I run Monolingual with the "Move language files to trash" preference selected.

1.  Monolingual 1.7.6 reports that it removed a very large number of files from /private/var/root/.Trash/
(Monolingual 1.7.6 also reports a much smaller number of files removed from ~/.Trash/)  The deleted files are listed in monolingual.log, which I've attached.  I've never seen this behavior before, and I find it rather alarming. What is going on? Should I be alarmed, or is this merely a UI issue?

2.  Monolingual 1.7.6 appears to list all deleted files in monolingual.log (not only deleted .lproj directories, but their contained files as well).  As a result, my monolingual.log is 5.5 MB in size after one run.  (I don't know if this is actually new behavior, of if Monolingual trims the previous list of deleted files from the log each time it runs.)

3.  Monolingual 1.7.6 is the first version to report disk space saved when moving language files to the trash, rather than deleting them in place.  (Previous versions always reported 0 MB saved.)  However, while Monolingual 1.7.6 reported over 1.6 GB saved, the Finder reports that the language files moved to the trash take up only 197 MB.

[Monolingual.log.zip](https://github.com/IngmarStein/Monolingual/files/759823/Monolingual.log.zip)

**Comment by IngmarStein** [comment]
1. That's a cosmetic issue in the log file. 1.7.6 prints the name of the file _after_ it has moved it to the trash. I'll fix that in the next version to print the original path instead.
2. That has always been the case. You can just delete the log if it grows too large. Each run appends to it, i.e. the file is not trimmed.
3. If it wasn't so much work to translate the string in that many languages, I would have changed the message to say that this is the space which will be saved when emptying the trash. I think that's better than reporting "0 bytes".

**Comment by IngmarStein** [comment]
Regarding the size difference: Monolingual first tries to move a file to the user's trash. If that fails (most likely due to permissions), it moves it to root's trash instead. If you just look at the user's trash, you're missing the other part.

**Comment by SeanMSmith** [comment]
Thank you for the quick response!  Re:

1.  I'm reassured to know that this is a cosmetic issue.
2.  If you look at the log file I appended, you'll notice that there are no lists of deleted files from runs of earlier versions of Monolingual (i.e. I did not trim the log by hand).  This does appear to be new behavior.  (It's not behavior that troubles me; I originally mentioned it in case it had some bearing on item 1.)
3.  I entirely concur that it's better to report space that will be saved when emptying the trash, rather than reporting 0 bytes, and I think the current wording of the message (at least in English) is good.  I always considered the previous behavior (of reporting 0 bytes saved) to be a bug. 

Regarding the size difference and the files moved to root's trash: pardon my ignorance, but how do I empty root's trash?

**Comment by IngmarStein** [comment]
1, 2 and 3 are related: Monolingual < 1.7.6 didn't report progress when moving files to the trash, so it didn't keep track of the sizes (leading to the "0 bytes" bug) and didn't write the paths to the log.

You can empty root's trash in Terminal: `sudo rm -rf /var/root/.Trash/*`.

**Comment by IngmarStein** [comment]
Does 1.7.7 fix the log file for you?

**Comment by SeanMSmith** [comment]
Re emptying root's trash, `sudo rm -rf /var/root/.Trash/*` did not work.  Instead I had to `sudo rm -rf /var/root/.Trash/` which of course removed the trash directory itself, which I then recreated with `mkdir -m 700 /var/root/.Trash`.

It turns out there were 4 GB of .lproj folders in /var/root/.Trash, far more than the 1.4 GB that Tuesday's run of Monolingual 1.7.6 had moved there!  How long the previous 2.6 GB of .lproj folders had been sitting there is anyone's guess.

I'll open a new issue with a request that Monolingual's UI should at least note that moving files to the trash before deleting them may result in many files being moved to root's trash, which the average user won't know about or want to tinker with.

Re Monolingual 1.7.7's log file, I observe the following changes:

a. The log file (and the application UI itself) reports the original .lproj directory location, not a location in the trash.  Yay!

b. While the log file no longer reports the names of deleted files, only the names of .lproj directories, it appears that each removed file is still being logged to a separate line, as the log file has multiple entries for many of the same .lproj directories, showing different file sizes, but with the filename apparently omitted. (I append my log file for your inspection.)
[Monolingual 1.7.7.log.zip](https://github.com/IngmarStein/Monolingual/files/765806/Monolingual.1.7.7.log.zip)

I note the more rapidly increasing log file size for the sake of completeness, but I'm not not fussed about it.  As far as I'm concerned, v1.7.7 has corrected the issues I initially reported.
